training/components.py

```diff
@@ -167,111 +169,44 @@ def list_phase1_final_model():
-                        requests:
-                          cpu: 8
-                          "nvidia.com/gpu": {nproc_per_node}
-                        limits:
-                          cpu: 8
-                          "nvidia.com/gpu": {nproc_per_node}
-                  volumes:
-                    - name: input-data
-                      persistentVolumeClaim:
-                        claimName: {input_pvc_name}
-                    - name: model
-                      persistentVolumeClaim:
-                        claimName: {model_pvc_name}
-                    - name: output
-                      persistentVolumeClaim:
-                        claimName: {output_pvc_name}
-            Worker:
-              replicas: {nnodes-1}
-              restartPolicy: OnFailure
-              template:
-                metadata:
-                  annotations:
-                    sidecar.istio.io/inject: 'false'
-                spec:
-                  containers:
-                    - args:
-                        - |
-                          echo "Running phase {phase_num}"
+    resources_per_worker = {"cpu": "8", "nvidia.com/gpu": nproc_per_node}
+
+    base_image = "quay.io/redhat-et/ilab:shrey"
+    name = f"train-phase-{phase_num}-{name_suffix.rstrip('-sdg')}"
+    command = ["/bin/bash", "-c", "--"]
+
+    master_args = [
+        f"""echo "Running phase {phase_num}"
+                        echo "Using {path_to_model} model for training"
+                        echo "Using {path_to_data} data for training"
+                        mkdir -p /output/phase_{phase_num}/model;
+                        mkdir -p /output/data;
+                        torchrun --nnodes {nnodes} \
+                            --nproc_per_node {nproc_per_node} \
+                            --node_rank \$(RANK) \
+                            --rdzv_endpoint \$(MASTER_ADDR):\$(MASTER_PORT) \
+                            -m instructlab.training.main_ds \
+                            --model_name_or_path={path_to_model} \
+                            --data_path={path_to_data} \
+                            --output_dir=/output/phase_{phase_num}/model \
+                            --num_epochs={num_epochs} \
+                            --effective_batch_size={effective_batch_size} \
+                            --learning_rate={learning_rate} \
+                            --num_warmup_steps={num_warmup_steps} \
+                            --save_samples={save_samples} \
+                            --log_level=INFO \
+                            --max_batch_len={max_batch_len} \
+                            --seed={seed} \
+                            --cpu_offload_optimizer \
+                            --cpu_offload_params \
+                            --distributed_training_framework fsdp \
+                            --is_granite \
+                            --checkpoint_at_epoch
+"""
+    ]
+
+    worker_args = [
```

Comment:
Same for `worker_args`, need to update the main_ds parameters to align with #230